### PR TITLE
WIP - documentation publication

### PR DESCRIPTION
**Pull request by rianfowler**
adds doc publication task to the jenkins pipeline

### CODE REVIEW

File: packages/documentation/package.json [hunk 1]
```diff
@@ -36,7 +37,7 @@
   ],
   "license": "MIT",
   "scripts": {
-    "build": "if [ -z \"$GITHUB_API_KEY\" ]; then echo \"No API key, skipping doc build\"; else gatsby build; fi",
+    "build": "if [ -z \"$GITHUB_API_KEY\" ]; then echo \"No API key, skipping doc build\"; else gatsby build --prefix-paths && gh-pages -d public -e ../..; fi",
```
Comment (rianfowler) [inline packages/documentation/package.json]: I verified that it pushes the build locally to the gh-pages [branch](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/tree/gh-pages/). I don't have access to the settings for this project to turn on https://department-of-veterans-affairs.github.io/veteran-facing-services-tools

File: Jenkinsfile [hunk 2]
```diff
@@ -40,21 +40,22 @@ pipeline {
...
+        sh 'git config --global user.name va-bot'
+        sh 'git config --global credential.helper "/bin/bash ' + env.WORKSPACE + '/scripts/credential-helper.sh"'
+        withCredentials([[
+          $class: 'UsernamePasswordMultiBinding',
+          credentialsId: 'va-bot',
+          usernameVariable: 'GIT_USERNAME',
+          passwordVariable: 'GIT_PASSWORD'
+        ]]) {
+          sh 'cd packages/documentation'
+          sh 'GITHUB_API_KEY=${env.TOKEN} npm run build'
```
Comment (rianfowler) [inline Jenkinsfile]: Not sure how or if I can test Jenkins locally but I uncommented this and updated the npm commands.

File: packages/documentation/package.json (hunk 1 above)
Comment (jbalboni) [inline packages/documentation/package.json]: I think `pathPrefix` needs to be added to gatsby-config.

Comment (jbalboni) [reply packages/documentation/package.json]: I can get this turned on when the DSVA admin person is back online this afternoon.

File: packages/documentation/README.md [hunk 3]
```diff
@@ -40,6 +36,22 @@ _Have another more specific idea? You may want to check out our vibrant collecti
...
+1. **Create a Github personal access token**
+
+   See instructions above
+
+1. **Run build and publish task**
+
+    Navigate to the documentation package and run.
+
+    ```
+    GITHUB_API_KEY=your-personal-access-token yarn build
```
Comment (rianfowler) [inline packages/documentation/README.md]: are we using yarn or npm in this project?

File: packages/documentation/gatsby-config.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ module.exports = {
     title: `VA.gov Developer's Documentations`,
     description: `Resources and documention for the Development within the VA.gov project`,
   },
+  pathPrefix: '/veteran-facing-services-tools',
```
Comment (rianfowler) [inline packages/documentation/gatsby-config.js]: This works- there are some fonts failing to load but I think we can address that separately.

File: packages/documentation/package.json (hunk 1 above)
Comment (rianfowler) [reply packages/documentation/package.json]: Looks good!

File: packages/documentation/README.md (hunk 3 above)
Comment (jbalboni) [reply packages/documentation/README.md]: Yarn

**Pull request merged**